File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        totalProduct = 1
        numZeros = 0
        retList = []
        for num in nums:
            if num:
                totalProduct *= num
            else:
                numZeros += 1
        
        if numZeros > 1:
            return [0] * len(nums)

        for num in nums:
            if numZeros:
                if num:
                    retList.append(0)
                else:
                    retList.append(totalProduct)
            else:
                retList.append(totalProduct // num)
        return retList

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        int longest = 0;
        for (int num : nums) {
            numSet.add(num);
        }

        for (int num : numSet) {
            if (!numSet.contains(num - 1)) {
                int tempLongest = 1;
                int currNum = num;

                while (numSet.contains(currNum + 1)) {
                    tempLongest++;
                    currNum++;
                }

                longest = Math.max(longest, tempLongest);
            }
        }
        return longest;
    }
}
